### PR TITLE
Refactor build code aka "The one in which we remove utils/go.mod"

### PR DESCRIPTION
**Pull request by DaveSys911**
Description:
Removes some dead code & adds polishing around ci to remove redudencies.
- Remove unused scripts & utils
- Remove build & test stages from Jenkins release job(redudent)
- Refactor hardcoded bash from gh actions into scripts hosted under `~/.ci/scripts` for consistent behaviour
- Remove `utils/*` directory

**Important** checks to see passing:
1. release pipeline https://internal-ci.elastic.co/job/cloudbeat/job/cloudbeat-mbp/job/PR-408
2. Pre-merge & build

### CODE REVIEW

**Comment by DaveSys911** [comment]
@oren-zohar Ready for review.

**Comment by oren-zohar** [comment]
@DaveSys911 how can I test it? what will it affect?

**Comment by DaveSys911** [comment]
> @DaveSys911 how can I test it? what will it affect? 

Already tested.
Jenkins pipeline & cloudbeat-ci build & pre-merge should path.

File: scripts/make/generate_notice.py [hunk 1]
```diff
@@ -18,7 +18,8 @@
 DEFAULT_BUILD_TAGS = "darwin,linux,windows"
```
Comment (oren-zohar) [inline scripts/make/generate_notice.py:18]: please revert changes here

File: scripts/make/generate_notice.py [hunk 2]
```diff
@@ -90,12 +96,14 @@ def go_license_detector(notice_out, deps_out, modules):
...
-        notice_template_file.write(beats_notice_template.replace("Elastic Beats", "Elastic Cloudbeat"))
+        notice_template_file = open(
+            os.path.join(tmpdir, "NOTICE.txt.tmpl"), "w")
+        notice_template_file.write(beats_notice_template.replace(
+            "Elastic Beats", "Elastic Cloudbeat"))
         notice_template_file.close()
 
         args = [
-            "go", "run", "-modfile=utils/go.mod", "go.elastic.co/go-licence-detector",
+            "go", "run", "-modfile=go.mod", "go.elastic.co/go-licence-detector",
```
Comment (oren-zohar) [inline scripts/make/generate_notice.py:106]: except this chage

**Review by oren-zohar — CHANGES_REQUESTED** [review]
please remove the autopep8

File: scripts/make/generate_notice.py (hunk 1 above)
Comment (DaveSys911) [reply scripts/make/generate_notice.py]: Done

**Comment by DaveSys911** [comment]
@oren-zohar added autopep conf file and set the "line too long" to ignore, so it won't run & flake8 will ignore it as well.

**Review by oren-zohar — APPROVED** [review]

**Pull request merged**